store/src/java/com/zimbra/cs/mailbox/ContactAutoComplete.java

```diff
@@ -764,7 +764,9 @@ public void addMatchedContacts(String query, Map<String, ? extends Object> attrs
         Mailbox mbox = MailboxManager.getInstance().getMailboxByAccountId(getRequestedAcctId());
         if (folderIDs == null) {
             for (Folder folder : mbox.getFolderList(octxt, SortBy.NONE)) {
-                if (folder.getDefaultView() != MailItem.Type.CONTACT || folder.inTrash()) {
+            	//added an extra condition in if block (ZBUG-2588)
+            	//it was never going into the Mountpoint condition
+            	if ((folder.getDefaultView() != MailItem.Type.CONTACT || folder.inTrash()) && !(folder instanceof Mountpoint)) {
```

Comment:
we can optimise conditions
if(mountpoint) {
} else if (contact view || trash) {
} else {
}

Reply:
Sure @silentsakky ... I will do